apps/web/modules/api/v2/auth/api-wrapper.ts

```diff
@@ -104,11 +106,12 @@ export const apiWrapper = async <S extends ExtendedSchemas>({
   }
 
   if (rateLimit) {
-    const rateLimitResponse = await checkRateLimitAndThrowError({
-      identifier: authentication.data.hashedApiKey,
-    });
-    if (!rateLimitResponse.ok) {
-      return handleApiError(request, rateLimitResponse.error);
+    const rateLimitResult = await checkRateLimit(rateLimitConfigs.api.v2, authentication.data.hashedApiKey);
+
+    if (!rateLimitResult.ok) {
+      logger.error("V2 API rate limit check failed", { error: rateLimitResult.error });
+    } else if (!rateLimitResult.data.allowed) {
+      return handleApiError(request, { type: "too_many_requests" });
     }
```

Comment:
🔧 **Fix**: We should use the helper function `applyRateLimit`, and we don't need to check for the `(!rateLimitResponse.ok)` because we already log this error in the `checkRateLimit` function.

It would be something like:

```suggestion
   if (rateLimit) {
    try {
      await applyRateLimit(rateLimitConfigs.api.v2, authentication.data.hashedApiKey);
    } catch {
      return handleApiError(request, {
        type: "too_many_requests",
      });
    }
  }
```